go/adbc/driver/snowflake/driver_test.go

```diff
@@ -36,6 +36,7 @@ import (
 
 	"github.com/apache/arrow-adbc/go/adbc"
 	"github.com/apache/arrow-adbc/go/adbc/driver/internal"
+	"github.com/apache/arrow-adbc/go/adbc/driver/snowflake"
```

Comment:
this is already imported as `driver` on the line below this, that's why the CI is failing

Reply:
Thanks @zeroshade. I had fixed that locally but have been dealing with another issue causing tests to fail. The bufferedwriter was tripping the checkedallocator test whereas the regular writer previously wasn't.

I was able to plug the original leak and have this all working but it will require a separate PR to arrow which I will push up. I did slightly expand scope here however because as a part of the testing I enabled the checked allocator for all snowflake tests and have been chasing down a few unrelated issues as well. I'll push up some commits so you can see exactly what I mean.

Reply:
thanks for chasing all this down

Reply:
Most tests should pass once [https://github.com/apache/arrow/pull/41698](https://github.com/apache/arrow/pull/41698) is merged and pulled in.

There is still just one test remaining with a memory leak, which is `TestNonIntDecimalLowPrecision`. It seems rely on compute kernels to cast decimal values to floats. This involves buffer preallocation in _some_ cases which doesn't appear to get released properly. I haven't gotten to the bottom of this yet but if anyone else is more familiar with it or has the time, I could use a second set of eyes.

Reply:
Hmm, running locally all the failures cleared up expect for the one I mentioned above. I took a look and it seems Github Actions is skipping the snowflake tests because the SNOWFLAKE_URI is not defined.

I'll have some time to look at it today but a second set of eyes would still be a great help.

Reply:
Ok I found the issue and pushed up the fix. We were missing a `Release` on a fairly narrow code path for certain precisions. All snowflake tests pass locally for me now.